flare_portal/experiments/forms.py

```diff
@@ -68,6 +71,67 @@ def save(self, *, experiment: Experiment) -> None:
         )
 
 
+class ParticipantUploadForm(forms.Form):
+    import_file = forms.FileField(
+        validators=[FileExtensionValidator(["csv"])],
+    )
+
+    @staticmethod
+    def open_csv(field):
+        # Open uploaded file
+        data = io.StringIO(field.read().decode("utf-8"))
+        return csv.DictReader(data)
+
+    def clean(self):
+        # Open uploaded file
+        upload = ParticipantUploadForm.open_csv(self.cleaned_data["import_file"])
+
+        # Build a list of ID's
+        pids = []
+        for row in upload:
+            if pid := row.get("pid"):
+                if len(pid) <= 24:
+                    pids.append(pid)
+                else:
+                    self.add_error(
+                        "import_file",
+                        "Participant ID's must be less than 25 characters.",
+                    )
+
+        # Validate and check if any ID's already exist
+        existing_participants = Participant.objects.filter(participant_id__in=pids)
+        if existing_participants.count() > 0:
+            self.add_error(
+                "import_file",
+                f"Particpant ID {existing_participants.first().participant_id} already exists.",
+            )
```

Comment:
Have to verify this behaviour. I think we talked about the system just ignoring duplicate IDs and creating the rest, but we should double check with the client if they would prefer to be told that there are duplicates.

Reply:
Has been changed to ignore duplicate ID's